tests/integration/api/controllers/place.spec.js

```diff
@@ -96,4 +97,94 @@ describe('Place API', () => {
       });
     });
   });
+
+  describe('GET /api/v1/place', async () => {
+    const getPage = Place.v1.getPage(dataContext);
+    const invalidContactType = 'invalidPlace';
+
+    it('returns a page of places for no limit and cursor passed', async () => {
+      const responsePage = await getPage(Qualifier.byContactType(placeType));
+      const responsePlaces = responsePage.data;
+      const responseCursor = responsePage.cursor;
+
+      expect(responsePlaces).excludingEvery(['_rev', 'reported_date']).to.deep.equalInAnyOrder([place0]);
+      expect(responseCursor).to.be.equal(null);
+    });
+
+    it(`throws error when user does not have can_view_contacts permission`, async () => {
+      const opts = {
+        path: `/api/v1/place`,
+        auth: { username: userNoPerms.username, password: userNoPerms.password },
+      };
+      await expect(utils.request(opts)).to.be.rejectedWith('403 - {"code":403,"error":"Insufficient privileges"}');
+    });
+
+    it(`throws error when user is not an online user`, async () => {
+      const opts = {
+        path: `/api/v1/place`,
+        auth: { username: offlineUser.username, password: offlineUser.password },
+      };
+      await expect(utils.request(opts)).to.be.rejectedWith('403 - {"code":403,"error":"Insufficient privileges"}');
+    });
+
+    it('throws 400 error when placeType is invalid', async () => {
+      const queryParams = {
+        'placeType': invalidContactType
+      };
+      const queryString = new URLSearchParams(queryParams).toString();
+      const opts = {
+        path: `/api/v1/place?${queryString}`,
+      };
+
+      await expect(utils.request(opts))
+        .to.be.rejectedWith(`400 - {"code":400,"error":"Invalid contact type [${invalidContactType}]."}`);
+    });
+
+    it('throws 400 error when limit is invalid', async () => {
+      const queryParams = {
+        placeType,
+        limit: -1
+      };
+      const queryString = new URLSearchParams(queryParams).toString();
+      const opts = {
+        path: `/api/v1/place?${queryString}`,
+      };
+
+      await expect(utils.request(opts))
+        .to.be.rejectedWith(`400 - {"code":400,"error":"The limit must be a positive number: [${-1}]."}`);
+    });
+
+    it('throws 400 error when cursor is invalid', async () => {
+      const queryParams = {
+        placeType,
+        cursor: '-1'
+      };
+      const queryString = new URLSearchParams(queryParams).toString();
+      const opts = {
+        path: `/api/v1/place?${queryString}`,
+      };
+
+      await expect(utils.request(opts))
+        .to.be.rejectedWith(
+          `400 - {"code":400,"error":"Invalid cursor token: [${-1}]."}`
+        );
+    });
+  });
+
+  describe('Place.v1.getAll', async () => {
+    // TODO: this test is a bit non-covering of the actual generator functionality
```

Comment:
@jkuester Using the `utils.` functions from above, I could not create multiple places of the same type to test the paging and generator functionality. Can I get some help on this?

Reply:
You should be able to build up new place objects like this:

```js
  const clinic1 = utils.deepFreeze(placeFactory.place().build({
    parent: {
      _id: place1._id,
      parent: {
        _id: place2._id
      }
    },
    type: 'clinic',
    contact: {}
  }));
```

Then, in the `before` function, just include those new objects in the array passed to `utils.saveDocs`.  


(Also, when I tested this, I realized that for some reason I had to change the assert in this test from using `excluding` to using `excludingEvery` (like you had already in the test above).